packages/pull-signal/test/interopService.test.ts

```diff
@@ -2,77 +2,295 @@ import { beforeAll, describe, expect, it } from "vitest";
 import { genericLogger } from "pagopa-signalhub-commons";
 
 import {
-  authorizedPurposeIdForPullSignals,
-  consumerAgreementDraftState,
-  dataPreparationForSignalConsumers,
+  createEservice,
+  createInteropContext,
+  createPurpose,
   dataResetForSignalConsumers,
-  eserviceIdPushSignals,
+  getAnEservice,
+  getAPurpose,
+  getUUID,
 } from "pagopa-signalhub-commons-test";
 import { operationPullForbidden } from "../src/model/domain/errors";
-
 import { config } from "../src/config/env.js";
 import { interopService, postgresDB } from "./utils";
 
 describe("PDND Interoperability service", () => {
   beforeAll(async () => {
     await dataResetForSignalConsumers(postgresDB, config.interopSchema);
-    await dataPreparationForSignalConsumers(postgresDB, config.interopSchema);
   });
 
-  it("should give permission to a signals consumer to pull a signal", async () => {
-    const purposeId = authorizedPurposeIdForPullSignals;
-    const eserviceId = eserviceIdPushSignals;
+  it("Should deny permission to a signal consumer without e-service, agreement, purpose", async () => {
+    const consumerId = getUUID();
+    const eserviceId = getUUID();
```

Comment:
I'll say: **"Should deny permission to a signal consumer without an agreement and purpose with an e-service"** . In that case e-service is not owned by consumer.